src/languages/stan.js

```diff
@@ -512,41 +410,74 @@ export default function(hljs) {
       hljs.HASH_COMMENT_MODE,
       BLOCK_COMMENT,
       {
-        // hack: in range constraints, lower must follow "<"
-        begin: /<\s*lower\s*=/,
-        keywords: 'lower'
+        scope: 'symbol',
+        match: /\s(pi|e|sqrt2|log2|log10)(?=\()/,
+        relevance: 0
+      },
+      {
+        match: regex.concat(/[<,]\s*/, regex.either(...RANGE_CONSTRAINTS), /\s*=/),
+        keywords: RANGE_CONSTRAINTS
       },
       {
-        // hack: in range constraints, upper must follow either , or <
-        // <lower = ..., upper = ...> or <upper = ...>
-        begin: /[<,]\s*upper\s*=/,
-        keywords: 'upper'
+        scope: 'keyword',
+        match: /\btarget\s*/,
       },
       {
-        className: 'keyword',
-        begin: /\btarget\s*\+=/
+        // highlights the 'T' in T[,] for only Stan language distributrions
+        match: [
+          /~\s*/,
+          regex.either(...DISTRIBUTIONS),
+          /(?:\(\))/,
+          /\s*T(?=\s*\[)/
+        ],
+        scope: {
+          2: "built_in",
+          4: "keyword"
+        }
       },
       {
-        begin: '~\\s*(' + hljs.IDENT_RE + ')\\s*\\(',
-        keywords: DISTRIBUTIONS
+        // highlights distributrions that end with special endings
+        scope: 'built_in',
+        match: /\w+(_lpdf|_lupdf|_lpmf|_cdf|_lcdf|_lccdf|_qf)(?=\s*[\(.*\)])/
       },
       {
-        className: 'number',
-        variants: [
-          {
-            begin: /\b\d+(?:\.\d*)?(?:[eE][+-]?\d+)?/
-          },
-          {
-            begin: /\.\d+(?:[eE][+-]?\d+)?\b/
-          }
+        // cannot ever have these endings with the ~
+        match: [
+          /~\s*/,
+          /\w+(_lpdf|_lupdf|_lpmf|_cdf|_lcdf|_lccdf|_qf)(?=\s*[\(.*\)])/
         ],
-        relevance: 0
+        scope: { 2: "illegal" }
```

Comment:
@highlightjs/core  Any interest in getting into illegal?  I know I've mentioned it in some thread somewhere, but we never have in the past and I'm thinking that now likely isn't the time either...

Reply:
I'm also ok if this just doesn't get highlighted. Do you have a suggestion for that and I can remove the illegal stuff?

Reply:
If it's not guarding another rule, just remove it, if it is then document what it's protecting against with a comment and remove the `scope` - which will cause it to be matched and consumed, but not highlighted.

Reply:
> @highlightjs/core Any interest in getting into illegal?

I don't think now's the time to get into this, this would be better suited as a separate issue or PR

Reply:
Ok, lets rip it out then. :)

Reply:
I've removed the `illegal` and changed to `no-highlight`. Let me know if that is the appropriate use. It is protecting against highlighting when there is a ~ and a distribution, that cannot ever happen and it is documented on line 439.